components/operations-controller/Makefile

```diff
@@ -0,0 +1,100 @@
+APP_NAME = compass-operations-controller
+APP_PATH = components/operations-controller
+BUILDPACK = eu.gcr.io/kyma-project/test-infra/buildpack-golang-toolbox:v20210125-6234473e
+SCRIPTS_DIR = $(realpath $(shell pwd)/../..)/scripts
+CRD_OPTIONS ?= "crd:trivialVersions=true"
+CHART_PATH = $(realpath $(shell pwd)/../..)/chart/compass/charts/operations-controller
+export GO111MODULE = on
+export SKIP_STEP_MESSAGE = "Do nothing for Go modules project"
+
+include $(SCRIPTS_DIR)/generic_make_go.mk
+VERIFY_IGNORE := /vendor\|/automock\|/testdata
+
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+all: manager
+
+# Build manager
+build-local:
+	go build -o bin/manager main.go
+
+# Run tests
+test-local: generate fmt vet manifests
+	go test ./... -coverprofile cover.out
+
+# Build manager binary
+manager: generate fmt vet
+	go build -o bin/manager main.go
+
+# Run against the configured Kubernetes cluster in ~/.kube/config
+run: generate fmt vet manifests
+	go run ./main.go
+
+# Install CRDs into a cluster
+install: manifests
+	kustomize build config/crd | kubectl apply -f -
+
+# Uninstall CRDs from a cluster
+uninstall: manifests
+	kustomize build config/crd | kubectl delete -f -
+
+# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
+deploy: manifests
+	cd config/manager && kustomize edit set image controller=${IMG_NAME}
+	kustomize build config/default | kubectl apply -f -
+
+# Generate manifests e.g. CRD, RBAC etc.
+manifests: controller-gen
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+
+# Copy the CRDs into the Compass chart.
+copy-crds-to-chart:
```

Comment:
We did `kustomize build config/default` only once in the beginning, we generated all yaml files, we populated them accordingly for our use case and copied them by hand into the component's Helm chart. We did that because we saw that only the CRD yaml files will change over time. The other yaml files are a deployment for the Manager that manages controllers which isn't going to change (and even if it needs to change, we'll directly make changes it in the Helm chart), some RBAC yamls regarding the deployment that aren't going to change also, and some prometheus stuff that aren't expected to change also. Only the image of the deployment is going to change which is currently implemented to be configurable.

We'll see how we can plug this copying of yaml files during the job that builds this installer and we'll consider options that replace the copying part.

Reply:
I was under the impression that everything should be generated with kustomize every time we deploy because the Makefile generated by kubebuilder has a `deploy` that does `kustomize build config/default | kubectl apply -f -`

Reply:
Good point. I think that they do this just to ensure that if there are changes to the manager deployment, for an example, they will catch them and apply them. I think that our case is different because we, mostly likely, won't have many changes to those non-CRD yaml files. And if we do, we can directly modify them in the component's Helm chart itself, not in the component's Kubebuilder project yaml files which serve mostly as a template.